Remove npm.js gitignore

Daksh14 opened the pull request.

Daksh14 committed:
Remove npm.js gitignore

Daksh14 committed:
Fix tests

Daksh14 committed:
Fix tests

ZER0 commented on npm.js:
```diff
@@ -0,0 +1,53 @@
+  package: {
+    // package.json properties
+    name: "@dusk-network/dusk-wallet-js",
+    version: Deno.args[0],
```
The idea here is the following:

1. If no version is given from the command line, it uses the most recent (in terms of semver) tag that is present. If no valid tags exists, throws an error.

2. If a version is given from the command line, it checks if it exists in the tags present in the repo. If not, throws an error.

In both cases, if the operation succeed, the tool needs to temporary checkout the code at the valid tag given or found, build the library, and then check back the initial starting commit.

If the repo is "dirty" – there are changes that are not committed – the tool should throw an error.

ZER0 reviewed (changes requested):
See the comment.
The tool / task requires more work that the code you used locally, because we *need* to create a coupling between the versioning of the node package and the tags of the repo; as mentioned yesterday.

That means that most likely you need to use a `semver` package for that, at least – you don't need a git one, most likely you should be able to execute everything by spawn a process and call `git` there. Of course, if `git` is not installed, you throw an error.

Daksh14 committed:
Add git tag support and check for commits

Daksh14 committed:
Add test for npm.js

Daksh14 committed:
Fix tests

Daksh14 committed:
Use semver for versions

Daksh14 committed:
Apply gitignore

Daksh14 committed:
Remove commented code

ZER0 committed:
Implement exu for loading and calling wallet-core

- Add `exu` library to manage WebAssembly module in abortable workers
- Update `.gitignore`
- Remove `mnemonic` related code from library
- Move deno tasks ('build', 'npm') under `.scripts` folder
- Remove `deno.lock` from the repository
- Remove `initSync`: `dusk_wallet_core.js` now exposes the bytecode
- Remove `jsonFromBytes` and `toBytes` from `wasm.js`
- Add `encoding` module with `encode`, `decode` and `parseEncodedJSON`
- Fix typo `getUnpsentNotes` to `getUnspentNotes`
- Change all the calls to WebAssembly module to be `async`
- Change `Wallet` code to use ES classes
- Remove `numberToLittleEndianByteArray`
- Add `u32toLE` using `DataView`
- Replace `arrayMax` with `maxBlockHeight`

Resolves #11

ZER0 committed:
Change `npm` library build test to be on its own

Daksh14 committed:
Update dusk_wallet_core.wasm asset

Split buffer and call owned function per notes

Daksh14 committed:
CI: Update env var

Daksh14 committed:
Git fetch before checking out

ZER0 commented on .github/workflows/test.yaml:
```diff
@@ -44,6 +44,8 @@ jobs:
     - name: Run Tests
       run: deno task test
+      env:
+        WALLET_CORE_PATH: "./assets/dusk_wallet_core.wasm"
```
The name is wrong, it should contains the version. Not sure why the asset now is stripped by the versioning.
Why you changed what I did yesterday without consulting? I thought I explain why it's needed. Please, rename it in `dusk_wallet_core@0.21.9-rc.wasm`, that was the name we had yesterday.

ZER0 commented on src/node.js:
```diff
@@ -20,6 +20,16 @@ import { path } from "../deps.js";
+    this.push(v);
+  }, this);
+};
+
```
That's _very_ bad: never do this, never modify the prototype of classes / objects you don't own: it's like the orphan rules of Rust we have to apply manually to JS.

Also, you can just use `concat` or the spread syntax.

ZER0 commented on src/node.js:
```diff
@@ -78,64 +88,22 @@ export async function sync(wasm, seed, node = NODE) {
+  const notes = owned.notes;
+  const nullifiers = owned.nullifiers;
+  const psks = owned.public_spend_keys;
+  const blockHeights = owned.block_heights.split(",").map(Number);
```
I think block_heights should be a `u64` so `Number` is not suitable.

ZER0 commented on tests/balance_test.js:
```diff
@@ -42,7 +42,7 @@ Deno.test({
   name: "25 psks",
   fn() {
-    assertEquals(psks.length, 9);
+    assertEquals(psks.length, 4);
```
I thought I said we should have 2 keys: any reason to use 4?

Daksh14 replied:
I kept it 4 because it won't be a significant difference between 2, I can change it to 2 if you want

ZER0 commented on assets/dusk_wallet_core.wasm:
Please, rename it as it was yesterday.

ZER0 commented on src/node.js:
```diff
@@ -60,6 +60,7 @@ export function StakeInfo(
  * @returns {Promise} Promise that resolves when the sync is done
  */
 export async function sync(wasm, seed, node = NODE) {
+  const startTime = performance.now();
```
I don't think those should be in production. It's probably a leftover from your testing. Please remove them.

ZER0 commented on src/node.js:
```diff
@@ -71,36 +72,78 @@ export async function sync(wasm, seed, node = NODE) {
-  const psks = owned.public_spend_keys;
-  const blockHeights = owned.block_heights.split(",").map(Number);
-  const lastPos = owned.last_pos;
+    let chunkSize = 632;
```
Where this number comes from? Also, it looks more like constant rather than a variable.

Why this changes was needed anyway? It doesn't looks like it's part of this PR and it's not documented anywhere.

ZER0 commented on src/node.js:
```diff
@@ -71,36 +72,78 @@ export async function sync(wasm, seed, node = NODE) {
+      psks.extend(ownedPsks);
+      blockHeights.extend(ownedBlockHeights);
+    }
+  }
```
See above.

Daksh14 replied:
This change came from wrongly merged branch, It should be reverted

ZER0 replied:
Perfect. As mentioned, since everything was merged in the same PR, I could only go commit by commit. So it's possible that a later commit reverted it. Please, as noted, in the future if we have a long living branch, do a separate branch to merge into it (but let's try to avoid long living branch to begin with…)

ZER0 commented on src/node.js:
```diff
@@ -109,14 +152,18 @@ export async function sync(wasm, seed, node = NODE) {
+  const endTime = performance.now();
+
+  console.log(`Sync took ${endTime - startTime} milliseconds`);
+
```
This looks like debug statements, the shouldn't be here.

ZER0 commented on tests/npm_build_test.js:
```diff
@@ -27,6 +27,12 @@ Deno.test({
+    });
+
+    await ensureSuccess(gitCommand.output());
+
```
This is wrong. Why are you fetching during a unit test? This might lead to several undesirable side effects.
Also, it's not documented anywhere *why* this is needed.

Daksh14 replied:
The CI wasn't passing without this, it was giving a pathspec - not found error. This fixed it

ZER0 replied:
Yeah, but:
a) Why it fixes it?
b) Where is the documentation about why it fixes it?
c) Have you considered what is the implication of doing so…?

ZER0 reviewed (changes requested):
I did the review commit after commit, however as mentioned previously for long living branch – that shouldn't exists to begin with, but sometimes this kind of things happen –we should branch out, having atomic commits that we will going to merge into the long living branch, and then to the default branch.
Here the changes are without atomic scope, there are not explained nor discussed, and therefore is basically impossible to follow the line of thoughts.

Daksh14 committed:
Fix regressions

Daksh14 committed:
Update CI

Daksh14 committed:
Remove npm build test (skipped)

Daksh14 committed:
Cleaning up long living branch

 - Remove console.*
 - Add `encodeStringifiedValue` to encoding.js
 - Change `JSON.stringify` to be executed inside `call`
 - Add comment explaining why we can't use bigint for block_height right now
